editor/modes/visual-editor/block.js

```diff
@@ -148,6 +167,8 @@ class VisualEditorBlock extends wp.element.Component {
 				onMouseMove={ this.maybeHover }
 				onMouseLeave={ onMouseLeave }
 				className={ className }
+				data-type={ block.blockType }
+				{ ...getBlockAttributesAsProps( block.attributes ) }
```

Comment:
Nice workaround :) 

I think we should propose an opt-in behaviour (maybe leave it undocumented until needed) and use it in core blocks. I'm a bit concerned with the too many `data-*` attributes we'll end up with.